lib/go-atscfg/remapdotconfig.go

```diff
@@ -289,16 +313,39 @@ func getServerConfigRemapDotConfigForMid(
 		if midRemap != "" {
 			midRemaps[remapFrom] = *ds.OrgServerFQDN + midRemap
 		}
+
+		// Any raw pre or post pend
+		dsPreRemaps, dsPostRemaps := lastPrePostRemapLinesFor(dsConfigParamsMap, *ds.XMLID)
+
+		// Add to pre/post remap lines if this is last tier
+		if 0 < len(dsPreRemaps) || 0 < len(dsPostRemaps) {
+			isLastCache, err := serverIsLastCacheForDS(server, &ds, nameTopologies, cacheGroups)
+			if err != nil {
+				warnings = append(warnings, "ds '"+*ds.XMLID+"' errors determining last tier "+err.Error())
```

Comment:
If this fails, it just keeps going and generates the line without the pre and post directives? I think the warning should probably say that, it's not clear from the warning that that happened.

Reply:
should it continue or fail?

Reply:
Looking at the code, it looks like the only way that function can fail, is if the data from TO is fundamentally broken.
It's things like
(1) The Server JSON has no CacheGroup
(2) A Topology entry references an index that doesn't exist
(3) A Topology references a CacheGroup name that doesn't exist.

It's all fundamentally-broken-data kind of stuff. I vote fail the config gen, return an error.